src/allotropy/parsers/lines_reader.py

```diff
@@ -40,8 +40,14 @@ def __init__(self, lines: list[str]) -> None:
         self.lines = lines
         self.current_line = 0
 
+    def line_exists(self, line: int) -> bool:
+        return 0 <= line < len(self.lines)
+
     def current_line_exists(self) -> bool:
-        return 0 <= self.current_line < len(self.lines)
+        return self.line_exists(self.current_line)
+
+    def get_line(self, line: int) -> Optional[str]:
```

Comment:
This method is unused -- please delete.

And if you're deleting this method, then there's really no need to factor out `line_exists` here, so I think you can undo all the changes in this file.